### PR TITLE
feat(mwc-button): support associated value

### LINKED ISSUE #2766: [mwc-button] missing value attribute
**Describe the bug**
`mwc-button` should feature `value` attribute and delegate its actual value to shadow tree `button` element to align with [native behavior](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-value).

**Expected behavior**
`mwc-button` submitting a form should define the value associated with the button’s name alongside form data.

**Additional context**
Not sure this should be a bug report or a feature request.
the reason is that button within shadow tree doesn't associate with form in the light tree, therefore supporting value attribute might not even work as expected until browsers support shadow tree form association out of the box.

#2761 sets the ground for this...

### PR DESCRIPTION
**Pull request by yinonov**
add missing `value` attribute as in https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-value

Fixes #2766

### CODE REVIEW

**Comment by yinonov** [comment]
Relates to #2759

**Comment by dfreedm** [comment]
Thanks for the PR!

File: packages/button/mwc-button-base.ts [hunk 1]
```diff
@@ -108,6 +111,7 @@ export class ButtonBase extends LitElement {
       <button
           id="button"
           class="mdc-button ${classMap(this.getRenderClasses())}"
+          .value="${live(this.value) as unknown as string}"
```
Comment (dfreedm) [inline packages/button/mwc-button-base.ts:114]: I don't think the `live` directive is doing much here, as the button value will not change on its own.

Let's remove it!

File: packages/button/mwc-button-base.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import {RippleHandlers} from '@material/mwc-ripple/ripple-handlers';
 import {eventOptions, html, LitElement, property, query, queryAsync, state, TemplateResult} from 'lit-element';
 import {ClassInfo, classMap} from 'lit-html/directives/class-map.js';
 import {ifDefined} from 'lit-html/directives/if-defined.js';
+import {live} from 'lit-html/directives/live.js';
```
Comment (dfreedm) [inline packages/button/mwc-button-base.ts:19]: Remove this import, since we won't need the `live` directive.

**Review by dfreedm — CHANGES_REQUESTED** [review]

**Comment by dfreedm** [comment]
Oh, and please rebase your PR to use the `lit` package for imports!

File: packages/button/mwc-button-base.ts (hunk 1 above)
Comment (yinonov) [reply packages/button/mwc-button-base.ts:114]: sure, keeping `as unknown as string` to align with textfield/textarea. not sure if required here

File: packages/button/mwc-button-base.ts [hunk 3]
```diff
@@ -13,9 +13,17 @@ import '@material/mwc-ripple/mwc-ripple';
...
+import {html, LitElement, TemplateResult} from 'lit';
+import {eventOptions, property, query, queryAsync, state} from 'lit/decorators.js';
+import {ClassInfo, classMap} from 'lit/directives/class-map.js';
+import {ifDefined} from 'lit/directives/if-defined.js';
+=======
 import {eventOptions, html, LitElement, property, query, queryAsync, state, TemplateResult} from 'lit-element';
 import {ClassInfo, classMap} from 'lit-html/directives/class-map.js';
 import {ifDefined} from 'lit-html/directives/if-defined.js';
+import {live} from 'lit-html/directives/live.js';
+>>>>>>> d200836fdaeb9ae4f48853fd6a9c1f068b0e76f8
```
Comment (yinonov) [inline packages/button/mwc-button-base.ts:26]:
```suggestion
import {html, LitElement, TemplateResult} from 'lit';
import {eventOptions, property, query, queryAsync, state} from 'lit/decorators.js';
import {ClassInfo, classMap} from 'lit/directives/class-map.js';
import {ifDefined} from 'lit/directives/if-defined.js';
```

File: packages/button/mwc-button-base.ts [hunk 4]
```diff
@@ -108,6 +118,11 @@ export class ButtonBase extends LitElement {
       <button
           id="button"
           class="mdc-button ${classMap(this.getRenderClasses())}"
+<<<<<<< HEAD
+          .value="${this.value as unknown as string}">
+=======
+          .value="${live(this.value) as unknown as string}"
+>>>>>>> d200836fdaeb9ae4f48853fd6a9c1f068b0e76f8
```
Comment (yinonov) [inline packages/button/mwc-button-base.ts:118]:
```suggestion
          .value="${this.value as unknown as string}">
```

**Comment by yinonov** [comment]
I purely `git rebase master` and it added all the commits and multiple files changes. Is it normal?

File: packages/button/mwc-button-base.ts [hunk 5]
```diff
@@ -108,6 +111,7 @@ export class ButtonBase extends LitElement {
       <button
           id="button"
           class="mdc-button ${classMap(this.getRenderClasses())}"
+          .value="${this.value as unknown as string}">
```
Comment (yinonov) [inline packages/button/mwc-button-base.ts:114]:
```suggestion
          .value="${this.value as unknown as string}"
```

**Comment by yinonov** [comment]
same change with cleaner commit history #2802. can close this 1

**Pull request closed without merging**